src/svc_auth_gss.c

```diff
@@ -660,13 +654,11 @@ _svcauth_gss(struct svc_req *req, bool *no_dispatch)
 	}
 gd_free:
 	mutex_unlock(&gd->lock);
-	if (gd_hashed) {
-		unref_svc_rpc_gss_data(gd);
-		gd_hashed = false;
-	}
+	unref_svc_rpc_gss_data(gd);
```

Comment:
We should hold a ref in a success case. The macro conversion to gotos lost that. We should only call unref in the case of a failure.